convokit/model/speaker.py

```diff
@@ -80,14 +80,21 @@ def name(self, new_name):
         self.id = new_name
 
     # End properties
+
+    def _check_utterance_id(self, ut_id: str):
+        """self.utterance_ids is List for storage reasons, but checking list membership is expensive when many utterances present,
+           so this convenience function runs a set conversion first (duplicates ok since we're just checking for membership)"""
+        utterances_set = set(self.utterance_ids)
+        return ut_id in utterances_set
```

Comment:
Why would creating a set on every call be faster? Creating a set is an O(n) operation, so it should be slower than just checking for membership in the list directly.

Reply:
You're right, this was my mistake. What I was originally intending to do was address the list membership op in `iter_utterances`, which takes place inside a loop over all utterances and thus results in an overall O(n^2) complexity. But then I noticed that `get_utterance` had a list membership check as well and tried to be too clever by addressing both in a single function, not realizing that by doing the set conversion inside that function I was just replicating the O(n) list membership check.

I've pushed a new commit that rolls back the change to `get_utterance` while updating `iter_utterances` with my originally intended fix, which does the set conversion _outside_ the loop so we end up with two O(n) operations instead of an O(n^2) one.